src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3533,6 +3545,39 @@ RequiresIntermediateRowPullUp(MultiNode *logicalPlanNode)
 }
 
 
+/*
+ * RequiresPushdown returns whether the expression can be pushed down to workers.
+ */
+static bool
+CanPushDownExpression(Node *expression,
+					  ExtendedOpNodeProperties *extendedOpNodeProperties)
+{
+	bool hasAggregate = contain_agg_clause(expression);
+	bool hasWindowFunction = contain_window_function(expression);
+	bool hasPushableWindowFunction =
+		hasWindowFunction && !extendedOpNodeProperties->hasNonPushableWindowFunction;
+
+	if (!hasAggregate && !hasWindowFunction)
+	{
+		return true;
+	}
+
+	/* aggregates inside pushed down window functions can be pushed down */
+	if (hasPushableWindowFunction)
+	{
+		return true;
+	}
+
+	if (hasAggregate && !hasWindowFunction &&
+		extendedOpNodeProperties->groupedByDisjointPartitionColumn)
+	{
+		return true;
+	}
+
```

Comment:
Note a future PR: it would be nice to not push down nextval(..) calls, since they fail when pushed down to the worker.